application/cms/models.py

```diff
@@ -1039,26 +1056,44 @@ class Measure(db.Model):
     reference = db.Column(db.String(32), nullable=True)  # optional internal reference
 
     # relationships
-    subtopics = db.relationship("Subtopic", secondary="subtopic_measure", back_populates="measures")
+    subtopics = db.relationship(
+        "Subtopic", secondary="subtopic_measure", back_populates="measures", order_by="asc(Subtopic.id)"
+    )
+    versions = db.relationship("MeasureVersion", back_populates="measure", order_by=desc(MeasureVersion.version))
 
     # Departmental users can only access measures that have been shared with them, as defined by this relationship
-    # TODO: Uncomment this and use back_populates (relationship declared both sides) once user_measure table exists
-    # shared_with = db.relationship(
-    #     "User",
-    #     lazy="subquery",
-    #     secondary=user_measure,
-    #     primaryjoin="Measure.id == user_measure.columns.measure_id",
-    #     secondaryjoin="User.id == user_measure.columns.user_id",
-    #     backref=db.backref("measures", lazy=True),
-    # )
-
-    # TODO: Uncomment these once MeasureVersion exists
-    # def get_versions(self):
-    #     return (
-    #         MeasureVersion.query.filter(MeasureVersion.measure_id == self.id)
-    #         .order_by(desc(MeasureVersion.version))
-    #         .all()
-    #     )
-    #
-    # def latest_published_version_id(self):
-    #     return self.get_versions()[0].id
+    shared_with = db.relationship(
+        "User",
+        lazy="subquery",
+        secondary="user_measure",
+        primaryjoin="Measure.id == user_measure.columns.measure_id",
+        secondaryjoin="User.id == user_measure.columns.user_id",
+        back_populates="measures",
+    )
+
+    @property
+    def has_published_version(self):
+        return any(mv.published for mv in self.versions)
+
+    @property
+    def latest_version(self):
+        return MeasureVersion.query.filter(MeasureVersion.measure_id == self.id, MeasureVersion.latest == True).one()
+
+    @property
+    def latest_published_version(self):
+        published_versions = MeasureVersion.query.filter(
+            MeasureVersion.measure_id == self.id, MeasureVersion.published == True
+        ).all()
+
+        if published_versions:
+            return max(published_versions, key=lambda x: x.version)
```

Comment:
I actually refactored this while you were reviewing to use the Python-side (numeric) ordering. Don't want to leave it as lexicographical ordering because it will eventually break.

Reply:
@samuelhwilliams we could probably coerce it to a float sql-side. Or change the actual database column type? The main advantage would be a slightly faster query (as only one row would need to be returned).

Reply:
This feels like a micro-optimisation?